### PR TITLE
Add breaking change for cellformatting event

### LINKED ISSUE #15361: [WinForms] `DataGridView.CellFormatting` event not raised if a tooltip is shown
## [Change Title]

To meet accessibility standards `DataGridView` now shows cell's text and errors as tooltips when navigated by both mouse and keyboard (.e.g tab/arrow navigation, hotkeys). 
NH5SSK7eId

### Version introduced

3.1

### Old behavior

Previously `DataGridView` would only show cell's tooltips and error text (if those were set) when navigated by mouse.
While editing a cell in a `DataGridView` that had `ShowCellToolTips = true`, `CellFormatting` event would be raised whenever other cells would be hovered and that did not have `ToolTip` property set.
otF94hrOSp

### New behavior

`DataGridView` now shows cell's text and errors as tooltips when navigated by both mouse and keyboard (.e.g tab/arrow navigation, hotkeys). 
While editing a cell in a `DataGridView` that had `ShowCellToolTips = true`, `CellFormatting` is no longer raised whenever other cells are hovered because the content of hovered cells is shown as tooltip.
ReproSteps gif

### Reason for change

Accessibility requirements

### Recommended action

Refactor the code that may have depended on `CellFormatting` event while `DataGridView` is in edit mode.

### Category

- Windows Forms

### Affected APIs

"Not detectable via API analysis"

---
#### Issue metadata

* Issue type: breaking-change

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #15361

### CODE REVIEW

File: includes/core-changes/windowsforms/cellformatting-event-not-raised.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+### CellFormatting event not raised if tooltip is shown
+
+To meet accessibility standards, a <xref:System.Windows.Forms.DataGridView> now shows a cell's text and errors as tooltips when navigated by both mouse and keyboard. If a tooltip is shown, the <xref:System.Windows.Forms.DataGridView.CellFormatting?displayProperty=nameWithType> event is not raised.
```
Comment (tdykstra) [inline includes/core-changes/windowsforms/cellformatting-event-not-raised.md]: Is "when navigated" generally accepted language for this?  It doesn't appear in the [style guide list of terms for mouse interactions](https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=29013) or in the [Describing interactions with UI article](https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=26472).   Is it referring to "when hovering over" with the mouse or "when selected" by the keyboard? My understanding of "navigating" to an element with a mouse would include selecting it, which is different from "hovering" over it. 

Should "both ... and ..." be "either ... or ..."?

Would it be accurate to change the last sentence to something like "If a tooltip is shown **for a cell**, the DataGridView.CellFormatting event is not raised **for other cells**."?

The "recommended action" section is brief and clear, could this summary be replaced by something like "The CellFormatting event is not raised while the DataGridView is in edit mode." and leave the details for the description section?

File: includes/core-changes/windowsforms/cellformatting-event-not-raised.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+#### Change description
+
+Previous to .NET Core 3.1, a <xref:System.Windows.Forms.DataGridView> would only show a cell's tooltips and error text when navigated by mouse. If the user edited a cell that had the <xref:System.Windows.Forms.DataGridView.ShowCellToolTips%2A> property set to `true`, the <xref:System.Windows.Forms.DataGridView.CellFormatting> event was raised whenever other cells that did not have <xref:System.Windows.Forms.DataGridView.ShowCellToolTips%2A> set were hovered over before the edit was committed.
```
Comment (tdykstra) [inline includes/core-changes/windowsforms/cellformatting-event-not-raised.md]: I'm finding the mix of "navigating," "hovering," and "editing" here and in the next paragraph hard to understand.  I don't know if it will be clear enough to people who are familiar with the API.

File: includes/core-changes/windowsforms/cellformatting-event-not-raised.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+Starting in .NET Core 3.1, a <xref:System.Windows.Forms.DataGridView> now shows a cell's text and errors as tooltips when navigated by both mouse and keyboard (for example, by using the Tab key, shortcut keys, or arrow navigation). While editing a cell that has the <xref:System.Windows.Forms.DataGridView.ShowCellToolTips%2A> property set to `true`, the <xref:System.Windows.Forms.DataGridView.CellFormatting> event is *not* raised when other cells are hovered over, because the content of the hovered cells is shown as a tooltip.
```
Comment (tdykstra) [inline includes/core-changes/windowsforms/cellformatting-event-not-raised.md]: > the CellFormatting event is *not* raised when other cells are hovered over, because the content of the hovered cells is shown as a tooltip.

Earlier it was stated that this ("the content of the hovered cells is shown as a tooltip") is dependent on the ShowCellToolTips setting in those other cells, here that condition isn't mentioned.

**Comment by gewarren** [comment]
Close/reopen to get clean build.

**Review by tdykstra — COMMENTED** [review]
I found this one hard to follow, left some thoughts for you to consider.

**Comment by gewarren** [comment]
@russkie @vladimir-krestov @Amy-Li02 Can you look this over to ensure I captured the nuances correctly?

https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/3.0-3.1?branch=pr-en-us-15955#cellformatting-event-not-raised-if-tooltip-is-shown

**Comment by gewarren** [comment]
Also, is there a plan to update the [ShowCellToolTips page](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.datagridview.showcelltooltips?view=netcore-3.0#System_Windows_Forms_DataGridView_ShowCellToolTips) for 3.1 to indicate that tooltips are shown not just by hovering but also by keyboard navigation?

**Comment by RussKie** [comment]
Thank you, LGTM 👍

**Comment by RussKie** [comment]
> Also, is there a plan to update the [ShowCellToolTips page](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.datagridview.showcelltooltips?view=netcore-3.0#System_Windows_Forms_DataGridView_ShowCellToolTips) for 3.1 to indicate that tooltips are shown not just by hovering but also by keyboard navigation?

We should. @vladimir-krestov could you please action it.

**Review by tdykstra — APPROVED** [review]
The revisions are helpful and resolve the issues, thanks!

**Comment by gewarren** [comment]
Thanks for the reviews @tdykstra @RussKie .
